widgets/common/src/labels/LabelsList.jsx

```diff
@@ -0,0 +1,55 @@
+const LabelPropType = PropTypes.shape({ key: PropTypes.string, value: PropTypes.string, isInSystem: PropTypes.bool });
+
+export const LabelsPropType = PropTypes.arrayOf(LabelPropType);
+
+export default function LabelsList({ labels, onChange }) {
+    const { Label, Icon, Popup } = Stage.Basic;
+    const maxLength = 20;
+    const newLabelColor = 'blue';
+
+    return (
+        <div className="ui multiple dropdown" style={{ paddingRight: '4.1em', minHeight: '2em' }}>
+            {_.map(labels, ({ key, value, isInSystem = true }) => {
+                const truncatedKey = _.truncate(key, { length: maxLength });
+                const truncatedValue = _.truncate(value, { length: maxLength });
+                const allowPopup = truncatedKey !== key || truncatedValue !== value;
+
+                return (
+                    <Popup key={`${key}:${value}`} open={allowPopup ? undefined : false} wide>
+                        <Popup.Trigger>
+                            <Label
+                                as="a"
+                                color={!isInSystem ? newLabelColor : undefined}
```

Comment:
I reckon the negation here is unnecessary, as we still need to provide values for both cases :smile: 

```suggestion
                                color={isInSystem ? undefined : newLabelColor}
```